sdk/metric/metricdata/metricdatatest/assertion.go

```diff
@@ -110,7 +110,7 @@ func IgnoreValue() Option {
 
 // AssertEqual asserts that the two concrete data-types from the metricdata
 // package are equal.
-func AssertEqual[T Datatypes](t *testing.T, expected, actual T, opts ...Option) bool {
+func AssertEqual[T Datatypes](t testing.TB, expected, actual T, opts ...Option) bool {
```

Comment:
Doing something like here: https://github.com/fluentassert/verify/blob/3122387c317e8b519e7766250978ed7de3f37dc0/failmsg.go#L8-L24 would be more extensible and may support more assertion libraries/frameworks.